int64[] s
int64[] g
int64[] t
int64[] time
int64[] theta
int64[] m
int64[] ID
---
int64 to
int64 from